Source/diablo.cpp

```diff
@@ -1965,11 +1770,9 @@ static void timeout_cursor(bool bTimeout)
  */
 void game_loop(bool bStartup)
 {
-	int i;
-
-	i = bStartup ? sgGameInitInfo.nTickRate * 3 : 3;
+	uint16_t i = bStartup ? sgGameInitInfo.nTickRate * 3 : 3;
 
-	while (i--) {
+	while ((i--) != 0) {

```

Comment:
Perhaps this would be more readable this way?

```suggestion
	while (i > 0) {
		i--;
```

Reply:
Indeed, i absolutely detest the other one.

Reply:
Now that I am awake I think that this could be better written as:
```cpp
	uint16_t wait = bStartup ? sgGameInitInfo.nTickRate * 3 : 3;

	for (unsigned i = 0; i < wait; i++) {
```